src/hotspot/share/runtime/safepoint.cpp

```diff
@@ -547,6 +547,20 @@ class ParallelCleanupTask : public WorkerTask {
     _do_lazy_roots(!VMThread::vm_operation()->skip_thread_oop_barriers() &&
                    Universe::heap()->uses_stack_watermark_barrier()) {}
 
+  uint expected_num_workers() const {
```

Comment:
should this be 'static' for internal linkage?

Reply:
It is a method and uses the non static member field `_do_lazy_roots`. Not a free function.

Reply:
oh ok, good.